cibyl/plugins/openstack/sources/zuul/actions.py

```diff
@@ -155,6 +113,105 @@ def get_target_jobs(self, **kwargs):
         return SpecArgumentHandler.Option.SPEC
 
 
+class DeploymentGenerator:
+    """Factory for generation of :class:`Deployment`.
+    """
+
+    class Tools:
+        """Tools the factory will use to do its task.
+        """
+        release_finder = ReleaseFinder()
+        """Takes care of finding the release of the deployment."""
+
+    def __init__(self, tools=Tools()):
+        """Constructor.
+
+        :param tools: The tools this will use.
+        :type tools: :class:`DeploymentGenerator.Tools`
+        """
+        self._tools = tools
+
+    def generate_deployment_for(self, variant, **kwargs):
+        """Creates a new deployment based on the data from a job's variant.
+
+        :param variant: The variant to fetch data from.
+        :type variant: :class:`cibyl.sources.zuul.transactions.VariantResponse`
+        :return: The deployment.
+        :rtype: :class:`Deployment`
+        """
+
+        def get_release():
+            if any(term in kwargs for term in ('spec', 'release')):
+                return release_finder.find_release_for(variant)
+
+            # Nothing means to ignore this field.
+            return ''
+
+        release_finder = self._tools.release_finder
+
+        return Deployment(
+            release=get_release(),
+            infra_type='',
+            nodes={},
+            services={}
+        )
+
+
+class DeploymentFiltering:
+    """Takes care of applying the filters coming from the command line to a
+    deployment.
+    """
+    Filter = Callable[[Deployment], bool]
```

Comment:
I don't get something. Shouldn't it be in the definition of a parameter or the return type of the method for example?

Reply:
Its below, on the constructor. The type of the 'filters' argument.